pkgs/development/tools/misc/micronucleus/default.nix

```diff
@@ -0,0 +1,38 @@
+{
+  pkgs
+, stdenv
+, libusb
+, fetchFromGitHub
+, lib
+}:
+stdenv.mkDerivation {
+   pname = "micronucleus";
+   version = "2.04";
+   src = fetchFromGitHub {
+     owner = "micronucleus";
+     repo = "micronucleus";
+     rev = "29317ea0c5b55b704a42ad318c52945cf3e2359e";
+     sha256 = "14msy9amlbflw5mqrbs57b7bby3nsgx43srr7215zyhfdgsla0in";
+   } + /commandline;
+
+   buildInputs = [ libusb ];
+
+   patchPhase = ''
+     sed -i 's\/usr/local/bin\'$out/bin/'\g' Makefile
+   '';
+
+   installPhase = ''
+     mkdir -p $out/bin
+     mkdir -p $out/lib/udev
+     cp micronucleus $out/bin
+     cp 49-micronucleus.rules $out/lib/udev
```

Comment:
Future work, not needed for this PR:

You might want to PR a makefile change upstream, which updates their install taget to use `DESTDIR` and installs the udev rule too.

We'd then only need to set `makeFlags = [ "DESTDIR=$(out)" ];`, and could drop the custom `installPhase` alltogether for the next release.